File: acm-java/src/main/java/leetcode/Solution1561.java
package leetcode;

import java.awt.*;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Solution1561 {
    public int maxCoins(int[] piles) {
        int i = 0, j = piles.length - 2;
        Arrays.sort(piles);
        int sum = 0;
        while (i < j) {
            sum += piles[j];
            i += 1;
            j -= 2;
        }
        return sum;
    }
}

File: acm-kotlin/src/main/kotlin/leetcode/Solution1561.kt
package leetcode

class Solution1561 {
    fun maxCoins(piles: IntArray): Int {
        return piles
            .sortedDescending()
            .filterIndexed { i, _ -> i % 2 == 1 }
            .take(piles.count() / 3)
            .sum()
    }
}